packages/input-datepicker/src/LionInputDatepicker.js

```diff
@@ -0,0 +1,413 @@
+   */
+  updated(c) {
+    super.updated(c);
+
+    if (c.has('errorValidators') || c.has('warningValidators')) {
+      const validators = [...(this.warningValidators || []), ...(this.errorValidators || [])];
+      this.__syncDisabledDates(validators);
+    }
+    if (c.has('label')) {
+      this.calendarHeading = this.calendarHeading || this.label;
+    }
+  }
+
+  _calendarOverlayTemplate() {
+    // TODO: try to improve 'this.__calendarConfig(tpl)' approach by creating and exposing lit
+    // directives
+    return html`
+      <lion-calendar-overlay-frame>
+        <span slot="heading">${this.calendarHeading}</span>
+        ${this.__calendarConfig(this._calendarTemplate())}
+      </lion-calendar-overlay-frame>
+    `;
+  }
+
+  /**
+  * Subclassers can replace this with their custom extension of
+  * LionCalendar, like `<my-calendar id="calendar"></my-calendar>`
+  */ // eslint-disable-next-line class-methods-use-this
+  _calendarTemplate() {
+    return html`
+      <lion-calendar id="calendar"></lion-calendar>
+    `;
+  }
+
+  /**
+   * This function is run right after the TemplateResult for calendar is created. Its goal is to
+   * abstract away all (private and protected) logic from the template, so that subclassers
+   * replacing templates 'survive' internal refactors and profit from feature upgrades.
+   */
+  __calendarConfig(templateResult) {
+    const node = renderAndGetFirstChild(templateResult);
+    Object.assign(node, this.__virtualCalendar);
+    node.selectedDate = this.__getSyncDownValue();
+    node.addEventListener('user-selected-date-changed', this._onCalendarUserSelectedChanged);
+    return node;
+  }
+
+  /**
+  * Subclassers can replace this with their custom extension invoker,
+  * like `<my-button><calendar-icon></calendar-icon></my-button>`
+  */ // eslint-disable-next-line class-methods-use-this
+  _invokerTemplate() {
+    return html`
+      <button>&#128197;</button>
+    `;
+  }
+
+  // Renders the invoker button + the calendar overlay invoked by this button
+  __createPickerAndReturnInvokerNode() {
+    const invokerNode = renderAndGetFirstChild(this._invokerTemplate());
+    invokerNode.id = this.__invokerId;
+    invokerNode.addEventListener('click', this.__openCalendarOverlay);
+    invokerNode.setAttribute('aria-haspopup', 'dialog');
+    invokerNode.setAttribute('aria-expanded', 'false');
+
+    this.localizeNamespacesLoaded.then(() => {
+      invokerNode.setAttribute(
+        'aria-label',
+        localize.msg('lion-input-datepicker:openDatepickerLabel'),
+      );
+      invokerNode.setAttribute('title', localize.msg('lion-input-datepicker:openDatepickerLabel'));
+    });
+
+    // TODO: ModalDialogController should be replaced by a more flexible
+    // overlay, allowing the overlay to switch on smaller screens, for instance from dropdown to
+    // bottom sheet (working name for this controller: ResponsiveOverlayController)
+    this._overlayCtrl = overlays.add(
```

Comment:
why protected?

Reply:
Let's have the discussion about this in the meeting about namings.
Idea is that it's available for subclassers